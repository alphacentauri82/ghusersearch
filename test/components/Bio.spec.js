import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';
import Bio from 'components/Bio';

describe('Component: Bio', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(
      <Bio
        text="This is some bio text with @link"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should not render if props is absent', () => {
    const component = renderer.create(
      <Bio />
    );
    expect(component).toMatchSnapshot();
  });

});
