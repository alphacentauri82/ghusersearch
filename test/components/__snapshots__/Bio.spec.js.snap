// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Bio should not render if props is absent 1`] = `null`;

exports[`Component: Bio should render 1`] = `
<p
  className="Bio_qh35s3"
  dangerouslySetInnerHTML={
    Object {
      "__html": "This is some bio text with <a href=\\"http://github.com/link\\">@link</a>",
    }
  }
/>
`;
